indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -166,6 +166,18 @@ public boolean createOrUpdateAndStartSupervisor(SupervisorSpec spec)
     }
   }
 
+  public boolean wasSupervisorSpecModified(SupervisorSpec spec)
+  {
+    Preconditions.checkState(started, "SupervisorManager not started");
+    Preconditions.checkNotNull(spec, "spec");
+    Preconditions.checkNotNull(spec.getId(), "spec.getId()");
+    Preconditions.checkNotNull(spec.getDataSources(), "spec.getDatasources()");
+    synchronized (lock) {
+      Preconditions.checkState(started, "SupervisorManager not started");
+      return metadataSupervisorManager.wasSupervisorSpecModified(spec);
```

Comment:
I don't think we should need to call metadata store here.
- If the supervisor is active (running or suspended), it would already be in memory in the map `SupervisorManager.supervisors`. We can compare with the version in memory to determine if the spec has been updated.
- If the supervisor is not present in the map, that indicates that the supervisor is either terminated or doesn't exist at all. In both the cases, we should just return `true`.